module/VuFind/src/VuFind/Db/Table/LoginToken.php

```diff
@@ -179,13 +179,18 @@ public function getBySeries(string $series): ResultSetInterface
     /**
      * Remove expired login tokens
      *
-     * @return void
+     * @param string   $dateLimit Date threshold of an "expired" record in format
+     * 'Y-m-d H:i:s'.
+     * @param int|null $limit     Maximum number of rows to delete or null for no
+     * limit.
+     *
+     * @return int
      */
-    public function deleteExpired(): void
+    public function deleteExpired($dateLimit, $limit = null): int
     {
         $callback = function ($select) {
             $select->where->lessThanOrEqualTo('expires', time());
         };
-        $this->delete($callback);
+        return $this->delete($callback);
```

Comment:
Login token expiration was never wired up correctly; I'm a bit surprised that the expiration command worked at all! I've adjusted things so it complies with the necessary interfaces, but this low-level code is currently ignoring the limits. I'm not sure how best to implement that, and I'm short on time -- I'd welcome follow-up work if anyone has time!

Reply:
I pushed a fix so that LoginToken now uses the standard expiration trait and just ignores the date threshold in the callback.

Reply:
Looks good to me (and works in my test environment). Thanks!